go-controller/pkg/ovn/policy.go

```diff
@@ -185,24 +191,40 @@ func hashedPortGroup(s string) string {
 	return util.HashForOVN(s)
 }
 
+func generatePrefix(s string) sets.String {
+	prefixes := sets.NewString()
+	for index := range s {
+		prefixes.Insert(s[:index])
+	}
+	prefixes.Insert(s)
+	return prefixes
+}
+
 // updateStaleDefaultDenyACLNames updates the naming of the default ingress and egress deny ACLs per namespace
 // oldName: <namespace>_<policyname> (lucky winner will be first policy created in the namespace)
 // newName: <namespace>_egressDefaultDeny OR <namespace>_ingressDefaultDeny
 func (oc *Controller) updateStaleDefaultDenyACLNames(npType knet.PolicyType, gressSuffix string) error {
 	cleanUpDefaultDeny := make(map[string][]*nbdb.ACL)
+	prefixes := generatePrefix(gressSuffix)
 	p := func(item *nbdb.ACL) bool {
+		aclNameSuffix := strings.Split(*item.Name, "_")
+		if len(aclNameSuffix) == 1 {
+			// doesn't have suffix; no update required; append the actual suffix since this ACL can be skipped
+			aclNameSuffix = append(aclNameSuffix, gressSuffix)
+		}
 		return item.ExternalIDs[defaultDenyPolicyTypeACLExtIdKey] == string(npType) && // default-deny-policy-type:Egress or default-deny-policy-type:Ingress
 			strings.Contains(item.Match, gressSuffix) && // Match:inport ==	@ablah80448_egressDefaultDeny or Match:inport == @ablah80448_ingressDefaultDeny
-			!strings.Contains(*item.Name, arpAllowPolicySuffix) && // != name: namespace_ARPallowPolicy
-			!strings.Contains(*item.Name, gressSuffix) // filter out already converted ACLs (we still continue to pick the cases where names were >45 chars)
+			!strings.Contains(item.Match, staleArpAllowPolicyMatch) && // != match: (arp || nd) OR arp
+			!prefixes.Has(aclNameSuffix[1]) // filter out already converted ACLs or ones that are a no-op
```

Comment:
we can just use here `strings.HasPrefix(gressSuffix, aclNameSuffix)`, we don't need `generatePrefix` I think

Reply:
the `HasPrefix` checks if the latter string provided is a prefix of the former, so the whole string would be matched as a prefix which in our case is the full `egressDefaultDeny`, we won't be accouting for the trimmed cases like `egress` or `egr` etc.. at least from reading the golang utility page for `HasPrefix` that is my understanding of how it works.

Reply:
How is that?
`strings.HasPrefix("egressDefaultDeny", "egress")` would be true?

> the HasPrefix checks if the latter string provided is a prefix of the former
correct

>  so the whole string would be matched as a prefix which in our case is the full egressDefaultDeny, we won't be accouting for the trimmed cases like egress or egr etc.. at least from reading the golang utility page for HasPrefix that is my understanding of how it works

no, in `strings.HasPrefix(gressSuffix, aclNameSuffix)`, `aclNameSuffix` is the (probably trimmed) suffix the ACL name has and `gressSuffix` is the full suffix constant so

the full case: `strings.HasPrefix("egressDefaultDeny", "egressDefaultDeny")` would be true as "egressDefaultDeny" is a prefix of "egressDefaultDeny"
the trimmed case: `strings.HasPrefix("egressDefaultDeny", "egress")` would be true as "egress" is a prefix of "egressDefaultDeny"

what am I missing?

Reply:
oh got it! sorry went off tangent